build: Bump rand and flate2

tesuji opened the pull request.

tesuji committed:
build: Bump rand to 0.7

kinnison reviewed (commented):
At first glance I don't think this will cause issues.  The `flate2` upgrade is trivial, the `rand` upgrade, while a new major version (0.6 to 0.7) looks compatible enough.

Does this reduce the complexity of our dependency tree at all?

kinnison commented:
I'm holding off on approving until we fix the spurious failures and then rebase.

tesuji committed:
build: Bump other deps

    Updating backtrace v0.3.30 -> v0.3.32
    Updating backtrace-sys v0.1.28 -> v0.1.30
    Updating chrono v0.4.6 -> v0.4.7
    Updating curl-sys v0.4.18 -> v0.4.19
    Updating digest v0.8.0 -> v0.8.1
    Updating futures v0.1.27 -> v0.1.28
    Updating flate2 v0.1.7 -> v0.1.9
    Updating generic-array v0.12.0 -> v0.12.3
    Updating getrandom v0.1.5 -> v0.1.6
    Updating h2 v0.1.23 -> v0.1.25
    Updating hyper v0.12.30 -> v0.12.31
    Updating ryu v0.2.8 -> v1.0.0
    Updating serde v1.0.92 -> v1.0.94
    Updating serde_derive v1.0.92 -> v1.0.94
    Updating serde_json v1.0.39 -> v1.0.40
    Updating string v0.2.0 -> v0.2.1
    Updating syn v0.15.36 -> v0.15.39
    Updating tempfile v3.0.8 -> v3.0.9
    Updating vcpkg v0.2.6 -> v0.2.7
    Updating want v0.0.6 -> v0.2.0
